### PR TITLE
Small fixes to achieve reproducible build

### PR DESCRIPTION
**Pull request by squadette**
hi,

I'm trying to achieve green tests on my machine.  So far I'm unsuccessful, but I've found few issues which I'd like to fix in master to help everyone else.

Thank you,

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/661)

### CODE REVIEW

**Comment by justin808** [comment]
Some suggestions

---

Reviewed 3 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[.gitignore, line 12 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpUpqxdumGObJslXB:-K_RpUpqxdumGObJslXC:b-wersto) ([raw file](https://github.com/shakacode/react_on_rails/blob/7d0f9e6e3dff6f0ecd6aa020f0b90943ac371dd6/.gitignore#L12)):*
> ```
> *.gem
> /vendor/
> *~
> ```

This is due to emacs probably and nothing to do with React on Rails. However, I'm OK with this if you think it's better here rather than your global gitignore.

---

*[CONTRIBUTING.md, line 89 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_Rp4uQuKebRqeh5Xh6:-K_Rp4uQuKebRqeh5Xh7:b-gr7l1e) ([raw file](https://github.com/shakacode/react_on_rails/blob/7d0f9e6e3dff6f0ecd6aa020f0b90943ac371dd6/CONTRIBUTING.md#L89)):*
> ```Markdown
> When you use a relative path, be sure to run the above `npm install` command whenever you change the node package for react-on-rails.
> 
> While we'd prefer to us `npm link`, we get errors. If you can figure out how to get `npm link react-on-rails` to work with this project, please file an issue or PR! This used to work with babel 5.
> ```

Since we're getting typos: `use` not `us`.

---

*[CONTRIBUTING.md, line 144 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpHagRYJITxOg-QX5:-K_RpHagRYJITxOg-QX6:bdk7ku9) ([raw file](https://github.com/shakacode/react_on_rails/blob/7d0f9e6e3dff6f0ecd6aa020f0b90943ac371dd6/CONTRIBUTING.md#L144)):*
> ```Markdown
>     alias: {
>       react: path.resolve('./node_modules/react'),
>       'react-dom': path.resolve('./node_modules/react-dom'),
> ```

Why is this needed?

---

*[CONTRIBUTING.md, line 166 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpJrA2i738I8YpyEG:-K_RpJrA2i738I8YpyEH:b-jp377f) ([raw file](https://github.com/shakacode/react_on_rails/blob/7d0f9e6e3dff6f0ecd6aa020f0b90943ac371dd6/CONTRIBUTING.md#L166)):*
> ```Markdown
> cd <top level>
> npm run dummy:spec
> ```
> ```

correct

---

*[lib/generators/react_on_rails/templates/base/base/client/package.json.tt, line 17 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpMyvp4bSChzKkxFc:-K_RpMyvp4bSChzKkxFd:bbuama2) ([raw file](https://github.com/shakacode/react_on_rails/blob/13cff3feb5b0364a84c26cf1d56f31ef963fd7d0/lib/generators/react_on_rails/templates/base/base/client/package.json.tt#L17)):*
> ```
>   "cacheDirectories": ["node_modules", "client/node_modules"],
>   "dependencies": {
>     "babel": "^6.5.2",
> ```

why is babel removed? We should be locking the version of it.

---

*[spec/dummy/client/package.json, line 12 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpT0sicLueJQqJK1X:-K_RpT0sicLueJQqJK1Y:bbuama2) ([raw file](https://github.com/shakacode/react_on_rails/blob/13cff3feb5b0364a84c26cf1d56f31ef963fd7d0/spec/dummy/client/package.json#L12)):*
> ```JSON
>   "dependencies": {
>     "autoprefixer": "^6.6.0",
>     "babel": "^6.5.2",
> ```

why is babel removed? We should be locking the version of it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-:-K_RpcDpZb5O9c7IaY9o:by8cv6w)*

**Comment by justin808** [comment]
@squadette Any update?

**Comment by squadette** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[.gitignore, line 12 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpUpqxdumGObJslXB:-K_t7gxIGt-5vrMB8nxp:b-mgawg0) ([raw file](https://github.com/shakacode/react_on_rails/blob/7d0f9e6e3dff6f0ecd6aa020f0b90943ac371dd6/.gitignore#L12)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

This is due to emacs probably and nothing to do with React on Rails. However, I'm OK with this if you think it's better here rather than your global gitignore.
</blockquote></details>

I'll setup my global gitignore, makes sense.

---

*[CONTRIBUTING.md, line 144 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpHagRYJITxOg-QX5:-K_t8FeefHHMQ-os2iBB:bw0yr1i) ([raw file](https://github.com/shakacode/react_on_rails/blob/7d0f9e6e3dff6f0ecd6aa020f0b90943ac371dd6/CONTRIBUTING.md#L144)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Why is this needed?
</blockquote></details>

I don't know.  But In the doc it says: 

*Side note: It's critical to use the alias section of the webpack config to avoid a double inclusion error. This has already been done for you in the example and dummy apps, but for reference:*

But in example and dummy apps it has both react and react-dom, while in .md it only mentions react.  I'd also like to understand why exactly we need this special treatment.

---

*[lib/generators/react_on_rails/templates/base/base/client/package.json.tt, line 17 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpMyvp4bSChzKkxFc:-K_t8bEDsaV77yWXhA7E:biw7j14) ([raw file](https://github.com/shakacode/react_on_rails/blob/13cff3feb5b0364a84c26cf1d56f31ef963fd7d0/lib/generators/react_on_rails/templates/base/base/client/package.json.tt#L17)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

why is babel removed? We should be locking the version of it.
</blockquote></details>

`babel` package is deprecated/no-op.   It only warns about the need to use `babel-cli`.

https://github.com/babel/babel/blob/master/packages/babel/cli.js

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/661)*

**Comment by justin808** [comment]
Please rebase on top of master. I'd like to create a release today.

**Comment by justin808** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[CONTRIBUTING.md, line 144 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpHagRYJITxOg-QX5:-K_uCNPyTOn660uqoJiP:b-r97442) ([raw file](https://github.com/shakacode/react_on_rails/blob/7d0f9e6e3dff6f0ecd6aa020f0b90943ac371dd6/CONTRIBUTING.md#L144)):*
<details><summary><i>Previously, squadette (Alexey Mahotkin) wrote…</i></summary><blockquote>

I don't know.  But In the doc it says: 

*Side note: It's critical to use the alias section of the webpack config to avoid a double inclusion error. This has already been done for you in the example and dummy apps, but for reference:*

But in example and dummy apps it has both react and react-dom, while in .md it only mentions react.  I'd also like to understand why exactly we need this special treatment.

</blockquote></details>

Good catch. I wonder if this is still needed. Safer to do this.

---

*[spec/dummy/client/package.json, line 12 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/661#-K_RpT0sicLueJQqJK1X:-K_uCXsYN07r7XIg2zUp:b3kxkuz) ([raw file](https://github.com/shakacode/react_on_rails/blob/13cff3feb5b0364a84c26cf1d56f31ef963fd7d0/spec/dummy/client/package.json#L12)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

why is babel removed? We should be locking the version of it.
</blockquote></details>

@squadette Any idea here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/661)*

**Comment by squadette** [comment]
Updated as per discussion.

Thanks,

File: lib/generators/react_on_rails/templates/base/base/client/package.json.tt [hunk 1]
```diff
@@ -14,7 +14,6 @@
   },
   "cacheDirectories": ["node_modules", "client/node_modules"],
   "dependencies": {
-    "babel": "^6.5.2",
```
Comment (justin808) [inline lib/generators/react_on_rails/templates/base/base/client/package.json.tt:17]: @squadette  why is babel removed?

Comment (squadette) [reply lib/generators/react_on_rails/templates/base/base/client/package.json.tt:17]: 'babel' is deprecated.

**Comment by justin808** [comment]
Thanks @squadette

**Pull request merged**